### PR TITLE
Add DeliveryCreateNoteCommand implementation to DeveloperGuide.md

### LINKED ISSUE #163: Update DG - Delivery Create Note

### PR DESCRIPTION
**Pull request by B-enguin**
Explained the current implementation of the DeliveryCreateNoteCommand in the DeveloperGuide Add activity and sequence diagrams for DeliveryCreateNoteCommand to support the explanation

Update DeliveryCreateNoteCommand.java and DeliveryStatusCommand.java to use `Model#getDelivery()` in accordance with Law of Demeter

Close #163

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -193,6 +193,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+2. The `LogicManager` calls the `AdressBookParser#parseCommand()` with `delivery note DELIVERY_ID --note NOTE`
+3. `AddressBookParser` will parse the command, and creates a new instance of `DeliveryCreateNoteCommandParser` calling
+   `DeliveryCreateNoteCommandParser#parse()` to parse the remaining input after the command word has been removed
+   (i.e. the command arguments)
+4. `DeliveryCreateNoteCommandParser` will parse the arguments, and return a new instance of `DeliveryCreateNoteCommand`
+   with the parsed `DELIVERY_ID` and `Note`
+5. `LogicManager` calls `DeliveryCreateNoteCommand#execute()`, first checking if the user is logged in by calling
+   `Model#getUserLoginStatus()`
+6. It thens attempt to fetch the `Delivery` with the specified `DELIVERY_ID`, and replaces that `Delivery` using
+   `Model#setDelivery()` with a newly created `Delivery` with identical fields except for its status which is
```
Comment (zhonghan721) [inline docs/DeveloperGuide.md]: Should it be note instead of status?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -193,6 +193,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+1. The user will specify a `Delivery` through its `id`. The user must specify a non-empty `Note` to replace the
+   current status of the selected delivery.
+2. If a number followed by the `Note` prefix followed by a non-empty string is not specified, or is in the incorrect
+   format, a `ParseException` will be thrown.
+3. If the command is parsed successfully, a new `DeliveryCreateNoteCommand` is created, and executed.
+4. If the user is not logged in during command execution, a `CommandException` will be thrown.
+5. The number provided is used to fetch the associated `Delivery` from `Model` if it exists. If the provided
+   number does not match any of the IDs of the `Delivery`s stored in `Model`, a `CommandException` is thrown.
+6. If the command completed successfully, the selected `Delivery` will be replaced by an identical `Delivery`
```
Comment (zhonghan721) [inline docs/DeveloperGuide.md]: Would "completes" be more suitable here?

File: docs/diagrams/DeliveryCreateNoteActivityDiagram.puml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    elseif (Check Duplicate Prefix) then ([Duplicate Prefix])
+        :ParseException: Duplicate Prefix;
+        stop
+    elseif (Parse ID) then ([Invalid ID])
+        :ParseException: Invalid Delivery ID;
+         stop
+    elseif (Parse Note) then ([Invalid Note])
+        :ParseException: Invalid Note;
+        stop
+    else ([Valid ID and Note Provided])
```
Comment (zhonghan721) [inline docs/diagrams/DeliveryCreateNoteActivityDiagram.puml]: Hmm, not sure if we are allowed to specify the conditions in the diamond

**Review by zhonghan721 — COMMENTED** [review]
Some general comments

File: docs/diagrams/DeliveryCreateNoteSequenceDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+deactivate DeliveryCreateNoteCommandParser
+'Hidden arrow to position the destroy marker below the end of the activation bar.
+DeliveryCreateNoteCommandParser -[hidden]-> AddressBookParser
+destroy DeliveryCreateNoteCommandParser
+
+AddressBookParser --> LogicManager : d
+deactivate AddressBookParser
+
+AddressBookParser --> LogicManager : d
+deactivate AddressBookParser
```
Comment (zhonghan721) [inline docs/diagrams/DeliveryCreateNoteSequenceDiagram.puml]: Is this a duplication?

**Review by zhonghan721 — COMMENTED** [review]
Some general comments

File: docs/diagrams/DeliveryCreateNoteSequenceDiagram.puml (hunk 4 above)
Comment (B-enguin) [reply docs/diagrams/DeliveryCreateNoteSequenceDiagram.puml]: Yep, good catch, will update accordingly

**Review by zhonghan721 — APPROVED** [review]
LGTM

**Pull request merged**